desktop/cmp/loadingindicator/LoadingIndicator.js

```diff
@@ -0,0 +1,116 @@
+ * A minimal / unobtrusive LoadingIndicator displaying an optional spinner and/or message to signal
+ * that a longer-running operation is in progress, without using a modal Mask. Can be explicitly
+ * shown or bound to a PendingTaskModel.
+ *
+ * Note that the Panel component's `loadingIndicator` prop provides a common and convenient way to
+ * add an indicator to a Panel without needing to manually create or manage this component.
+ */
+@HoistComponent
+export class LoadingIndicator extends Component {
+
+    static modelClass = PendingTaskModel;
+
+    static propTypes = {
+
+        /** Position of the indicator relative to its containing component. */
+        corner: PT.oneOf(['tl', 'tr', 'bl', 'br']),
+
+        /** True to display the indicator. */
+        isDisplayed: PT.bool,
+
+        /**  Max characters allowed in message, after which it will be elided. Default 30. */
+        maxMessageLength: PT.number,
+
+        /** Optional text to be displayed - can also be sourced from bound PendingTaskModel. */
+        message: PT.string,
+
+        /** Optional model for reactively showing the indicator while tasks are pending. */
+        model: PT.instanceOf(PendingTaskModel),
+
+        /** Click handler. **/
+        onClick: PT.func,
+
+        /** True (default) to display with an animated spinner. */
+        spinner: PT.bool
+
+    };
+
+    baseClassName = 'xh-loading-indicator';
+    spinnerSize = 20;
+
+    render() {
+        const {props, model, message, showSpinner, corner} = this,
+            isDisplayed = withDefault(props.isDisplayed, model && model.isPending, false);
+
+        if (!isDisplayed || (!showSpinner && !message)) return null;
+
+        const hasMessageCls = message ? 'xh-loading-indicator--has-message' : null,
+            hasSpinnerCls = showSpinner ? 'xh-loading-indicator--has-spinner' : null,
+            cornerCls = `xh-loading-indicator--${corner}`;
+
+        return div({
+            className: this.getClassName(hasMessageCls, hasSpinnerCls, cornerCls),
+            onClick: props.onClick,
+            item: hbox(this.renderInnerItems())
+        });
+    }
+
+
+    //------------------------
+    // Implementation
+    //------------------------
+    renderInnerItems() {
+        const {corner, message, showSpinner} = this,
+            spinnerEl = spinner({size: this.spinnerSize});
+
+        if (!message) return [spinnerEl];
+
+        const msgBox = div({
+            className: `${this.baseClassName}__message`,
+            item: message
+        });
+
+        switch (corner) {
+            case 'tl':
+            case 'bl': return [showSpinner ? spinnerEl : null, msgBox];
+            case 'tr':
+            case 'br': return [msgBox, showSpinner ? spinnerEl : null];
+        }
+    }
+
+    get message() {
+        const {maxMessageLength, message, model} = this.props;
+        return truncate(withDefault(message, model && model.message), {length: withDefault(maxMessageLength, 30)});
+    }
+
```

Comment:
Would make maxMessageLength a getter, like the other two below, for symmetry

Reply:
Done.